plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/contributors/FirCallableCompletionContributor.kt

```diff
@@ -170,6 +173,73 @@ internal open class FirCallableCompletionContributor(
         collectTopLevelExtensionsFromIndices(listOf(typeOfPossibleReceiver), extensionChecker, visibilityChecker)
             .filter { filter(it) }
             .forEach { addCallableSymbolToCompletion(expectedType, it, getOptions(it)) }
+        if (explicitReceiver is KtSuperExpression) {
```

Comment:
I thought about that but that would require duplicating some logic setting up the explicit receiver and collecting members on the receiver. Since the entries provided by this logic are also "callables", it seems putting it here is natural. Let me know if you really think we should move this logic out.

Reply:
It is better to separate those contributors. FE1.0 completion suffers from a lot of mixed code. So it is better to separate different completion kinds. The way to avoid code duplication is introducing `super` completion contributor as an inheritor of `FirCallableCompletionContributor` like it is done for `FirCallableReferenceCompletionContributor`/`FirInfixCallableCompletionContributor`

Reply:
Hmm, this would not work very well because the super completion is in the same completion context with `FirCallableCompletionContributor`. Inheriting from `FirCallableCompletionContributor` could easily cause the same logic run twice if one is not careful. Let me extract utility function to share common logic instead.